src/rpc/util.cpp

```diff
@@ -793,7 +794,7 @@ void RPCResult::ToSections(Sections& sections, const OuterType outer_type, const
         return;
     }
     case Type::ANY: {
-        CHECK_NONFATAL(false); // Only for testing
+        NONFATAL_UNREACHABLE(); // Only for testing
     }
```

Comment:
This will also fix a warning on current master with gcc-12:


```
In file included from ./rpc/util.h:19,
                 from rpc/util.cpp:8:
rpc/util.cpp: In member function 'void RPCResult::ToSections(Sections&, OuterType, int) const':
./util/check.h:34:9: warning: this statement may fall through [-Wimplicit-fallthrough=]
   34 |         if (!(condition)) {                                       \
      |         ^~
rpc/util.cpp:796:9: note: in expansion of macro 'CHECK_NONFATAL'
  796 |         CHECK_NONFATAL(false); // Only for testing
      |         ^~~~~~~~~~~~~~
rpc/util.cpp:798:5: note: here
  798 |     case Type::NONE: {
      |     ^~~~